pkgs/development/python-modules/orjson/default.nix

```diff
@@ -0,0 +1,38 @@
+{ lib
+, buildPythonPackage
+, maturin
+, fetchFromGitHub
+, wheel
+, toml
+}:
+
+buildPythonPackage rec {
+  pname = "orjson";
+  version = "3.4.1";
+
+  src = fetchFromGitHub {
+    owner = "ijl";
+    repo = pname;
+    rev = version;
+    sha256 = "2+CZ4K4Sijt3ksjD6pVteRNNf+4z0iwZQon11b0+nyk=";
+  };
+
+  format = "pyproject";
+
+  nativeBuildInputs = [
+    maturin
+    wheel
+    toml
+  ];
+
+  meta = {
+    description = "Fast, correct Python JSON library supporting dataclasses, datetimes, and numpy";
+    license = with lib.licenses; [ 
+      asl20
+      mit
+    ];
+    homepage = "https://github.com/ijl/orjson";
+    broken = true; # Requires rust nightly
```

Comment:
I don't see how that can work even if that hadn't required rust nightly -- have you tried on omething that builds with maturein and a common rust?
I mean it sounds like it'd require internet or vendoring cargo deps like we do in other rust packages; otherwise maturin would just let cargo try to pull its deps as it feels like?

Reply:
If cargo needs to fetch dependencies then indeed it won't work. We basically need parts of `buildRustPackage` then.

Reply:
orjson's Cargo.toml:
```
[dependencies]
associative-cache = { version = "1" }
bytecount = { path = "./bytecount", default_features = false, features = ["generic-simd", "runtime-dispatch-simd"] }
encoding_rs = { path = "./encoding_rs", default_features = false, features = ["simd-accel"] }
inlinable_string = { version = "0.1" }
itoa = { version = "0.4", default_features = false }
once_cell = { version = "1", default_features = false }
pyo3 = { version = "0.12", default_features = false, features = ["extension-module"]}
ryu = { version = "1" }
serde = { version = "1", default_features = false }
serde_json = { path = "./json", default_features = false, features = ["std"] }
smallvec = { version = "1", default_features = false, features = ["const_generics", "union", "specialization", "write"] }
wyhash = { version = "0.4" }
```

I think it would, yes :)
Honestly they'd all reqiure at least pyo3 so it's unrealistic without; that's the reason I ended up using buildRustPackage instead of the python variant for ankirspy. I haven't looked at how to mix & match both, kind of lost motivation now I've seen this requires (and will keep requiring for the foreseeable future) nightly :/